File: packages/engine/src/cpus/6502/instructions/clc.js
import { CycleType } from '..'

/**
 * Clear Carry
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       CLC           $18  1   2
 */
export function* clc() {
  this.carry = false
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/clc.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Clear Carry
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       CLC           $18  1   2

pub fn clc(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.c = false;
}
